Koronavilkku/Koronavilkku/UI/Components/Checkbox.swift

```diff
@@ -0,0 +1,90 @@
+import SnapKit
+import UIKit
+
+class Checkbox: UIView {
+    enum Text: String, Localizable {
+        case AccessibilityValueUnchecked
+        case AccessibilityValueChecked
+        case AccessibilityHint
+    }
+    
+    private let acceptButton = UIButton()
+    private let labelView: UILabel
+    private let tapped: (Bool) -> ()
+    
+    private(set) var labelStartConstraint: ConstraintItem!
+    
+    var isChecked: Bool {
+        acceptButton.isSelected
+    }
+    
+    init(label: String, tapped: @escaping (Bool) -> ()) {
+        self.tapped = tapped
+        
+        self.labelView = UILabel(label: label,
+                                 font: UIFont.bodySmall,
+                                 color: UIColor.Greyscale.darkGrey)
+        
+        super.init(frame: .zero)
+
+        acceptButton.isSelected = false
+        acceptButton.addTarget(self, action: #selector(toggleTapped), for: .touchUpInside)
+        acceptButton.backgroundColor = .clear
+        acceptButton.layer.cornerRadius = 5
+        acceptButton.layer.borderWidth = 2
+        acceptButton.layer.borderColor = UIColor.Primary.blue.cgColor
+        acceptButton.setBackgroundImage(UIImage(named: "check")!.withTintColor(UIColor.Greyscale.white), for: .selected)
+        acceptButton.imageView?.contentMode = .scaleAspectFit
+        acceptButton.backgroundColor = acceptButton.isSelected ?
+            UIColor.Primary.blue :
+            .clear
+        
+        self.addSubview(acceptButton)
+
+        acceptButton.snp.makeConstraints { make in
+            make.top.equalToSuperview().offset(20)
+            make.left.equalToSuperview().offset(20)
+            make.width.height.equalTo(22)
+            make.bottom.lessThanOrEqualToSuperview().offset(20)
+        }
+        
+        labelView.numberOfLines = 0
+        self.addSubview(labelView)
+        labelStartConstraint = labelView.snp.left
+        labelView.snp.makeConstraints { make in
+            make.top.equalToSuperview().offset(20)
+            make.left.equalTo(acceptButton.snp.right).offset(10)
+            make.right.equalToSuperview().offset(-20)
+            make.bottom.lessThanOrEqualToSuperview().offset(-20)
+        }
+
+        let tap = UITapGestureRecognizer(target: self, action: #selector(toggleTapped))
+        labelView.addGestureRecognizer(tap)
```

Comment:
I think the touch/tap area has decreased: earlier the entire 'checkbox' area was registered for taps, whereas now only the acceptButton+label handle those? Before looking at the code while testing the checkbox it just felt a tad bit difficult to toggle the checked state.

Reply:
I was experimenting with different ways of handling the checkbox accessibility and this was required to make sure the tap recogniser is not run when user attempts to activate the element. As this is not required anymore, I reverted the behaviour.